### PR TITLE
add: order unit test, refix: connect robot

### PR DESCRIPTION
**Pull request by huuthiet**

### CODE REVIEW

File: app/order-api/src/tracking/tracking.service.ts [hunk 1]
```diff
@@ -286,4 +318,187 @@ export class TrackingService {
...
+  @Cron(CronExpression.EVERY_30_SECONDS, { name: 'UpdateStatusTracking' })
+  async UpdateStatusTracking() {
+    const context = `${TrackingService.name}.${this.UpdateStatusTracking.name}`;
+    const trackings = await this.trackingRepository.find({
+      where: {
+        id: this.currentTrackingUpdate,
+        status: In([WorkflowStatus.PENDING, WorkflowStatus.RUNNING]),
+      }
+    });
+    if(trackings.length < 1) this.stopCronJob('UpdateStatusTracking');
```
Comment (tuanhcmute) [inline app/order-api/src/tracking/tracking.service.ts:372]: Use lodash to check empty array

File: app/order-api/src/tracking/tracking.service.ts [hunk 2]
```diff
@@ -286,4 +318,187 @@ export class TrackingService {
...
+      throw new BadRequestException('Create tracking adn tracking order item failed')
+    } finally {
+      await queryRunner.release();
+    }
+  }
+
+  /**
+   * Update status of tracking every 30 seconds 
+   */
+  @Cron(CronExpression.EVERY_30_SECONDS, { name: 'UpdateStatusTracking' })
```
Comment (tuanhcmute) [inline app/order-api/src/tracking/tracking.service.ts:363]: Make 'UpdateStatusTracking' to constant

File: app/order-api/src/tracking/tracking.service.ts [hunk 3]
```diff
@@ -286,4 +318,187 @@ export class TrackingService {
...
+      })
+    );
+    
+  }
+
+  /**
+   * Stop a specific cron function
+   * @param {string} name The name of cron function
+   */
+  private stopCronJob(name: string) {
```
Comment (tuanhcmute) [inline app/order-api/src/tracking/tracking.service.ts:414]: update name to 'stopTrackingStatusCronJob'

File: app/order-api/src/tracking/tracking.service.ts [hunk 4]
```diff
@@ -286,4 +318,187 @@ export class TrackingService {
...
+        Object.assign(order, { status: OrderStatus.SHIPPING });
+        const updatedOrder = await this.orderRepository.save(order);
+        defaultStatus = updatedOrder.status;
+      }
+    } else if (totalBase === totalQuantities[WorkflowStatus.COMPLETED]) {
+      Object.assign(order, { status: OrderStatus.COMPLETED });
+      const updatedOrder = await this.orderRepository.save(order);
+      defaultStatus = updatedOrder.status;
+    }
+    console.log('updated in order function')
```
Comment (tuanhcmute) [inline app/order-api/src/tracking/tracking.service.ts:502]: use logger instead console.log

File: app/order-api/src/tracking/tracking.service.ts [hunk 5]
```diff
@@ -9,16 +9,24 @@ import { CreateTrackingRequestDto, TrackingResponseDto } from "./tracking.dto";
...
+import { ConfigService } from "@nestjs/config";
+import { RobotStatus } from "src/robot-connector/robot-connector.constants";
+import { Cron, CronExpression, SchedulerRegistry } from "@nestjs/schedule";
 
 @Injectable()
 export class TrackingService {
+  private readonly robotId: string =
+    this.configService.get<string>('ROBOT_ID');
+
+  private currentTrackingUpdate: string = null; //id
```
Comment (tuanhcmute) [inline app/order-api/src/tracking/tracking.service.ts:28]: remove currentTrackingUpdate var

File: app/order-api/src/order-item/order-item.dto.ts [hunk 6]
```diff
@@ -50,16 +50,16 @@ export class UpdateOrderItemRequestDto {
...
-  [WorkFlowStatus.RUNNING]: number;
+  [WorkflowStatus.RUNNING]: number;
 
   @AutoMap()
-  [WorkFlowStatus.COMPLETED]: number;
+  [WorkflowStatus.COMPLETED]: number;
 
   @AutoMap()
-  [WorkFlowStatus.FAILED]: number;
+  [WorkflowStatus.FAILED]: number;
```
Comment (tuanhcmute) [inline app/order-api/src/order-item/order-item.dto.ts:62]: dont use [WorkflowStatus.FAILED]

File: app/order-api/src/robot-connector/robot-connector.client.ts [hunk 7]
```diff
@@ -54,13 +77,13 @@ export class RobotConnectorClient {
...
     const requestUrl = `${this.robotApiUrl}/workflows`;
     const { data } = await firstValueFrom(
-      this.httpService.get<WorkFlowResponseDto[]>(requestUrl).pipe(
+      this.httpService.get<WorkflowResponseDto[]>(requestUrl).pipe(
         catchError((error: AxiosError) => {
           this.logger.error(
-            `Get all WorkFlows from ROBOT API failed: ${error.message}`,
+            `Get all Workflows from ROBOT API failed: ${error.message}`,
           );
           throw error;
```
Comment (tuanhcmute) [inline app/order-api/src/robot-connector/robot-connector.client.ts:88]: use throw new BadException instead  throw error;

**Review by tuanhcmute — CHANGES_REQUESTED** [review]

File: app/order-api/src/robot-connector/robot-connector.client.ts (hunk 7 above)
Comment (huuthiet) [reply app/order-api/src/robot-connector/robot-connector.client.ts:88]: not use now

**Review by tuanhcmute — APPROVED** [review]

**Pull request merged**